docs/source/cpp/api/array.rst

```diff
@@ -110,3 +110,9 @@ Utilities
    :project: arrow_cpp
    :members:
    :undoc-members:
+
+.. _api-array-from-json:
+
+.. doxygengroup:: array-from-json
```

Comment:
Is this section ill-formed? I saw "group array-from-json" rendered out in the output document.
<img width="758" alt="Screenshot 2025-04-21 at 13 48 15" src="https://github.com/user-attachments/assets/be30cd8c-9ec4-4ee4-88f7-9d4071ac97d8" />

Reply:
I'm not sure, I haven't used this directive before. I'll see if there are any other ways to do this bit.

Reply:
I see most of other occurrences of `doxygengroup` are like:
```
.. doxygengroup:: numeric-arrays
   :content-only:
   :members:
```
(They don't have a `:project:` one). That worth trying?

Reply:
I remember why I did it this way now. I originally did this with `content-only` but didn't like that it smushed all the helpers together with the Visit helpers (I wanted more structure). 

In 8b38e857f4bf48e07e735338fb0456b485ad84b8 I just added a sub-section and restored `content-only` so this renders better now:

![Screenshot 2025-04-21 at 3 31 00 PM](https://github.com/user-attachments/assets/d64abc53-97b2-4999-886f-41d2be59eb88)

Reply:
This looks much better. Thanks!